### PR TITLE
Jc automation more tests

### PR DESCRIPTION
**Pull request by jdcanas**
More tests!

Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookExtensionSpec.scala [hunk 1]
```diff
@@ -87,13 +88,14 @@ class NotebookExtensionSpec extends ClusterFixtureSpec with NotebookTestUtils {
               notebookPage.addCodeAndExecute("1+1")
 
               notebookPage.saveNotebook()
-              //sleep 4 minutes. We do this to ensure the assertions are true after a certain about of time
+
+              //sleep 4 minutes. We do this to ensure the assertions are true after a certain about of time.
+              //eventually is not applicable as we want to ensure its true after the time period, not at some point until timeout
               logger.info("Waiting 4 minutes as lock takes time to be reflected in metadata")
               Thread.sleep(240000)
```
Comment (Qi77Qi) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookExtensionSpec.scala:95]: can we remove this sleep now?

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/Notebook.scala [hunk 2]
```diff
@@ -45,6 +48,16 @@ object Notebook extends RestClient with LazyLogging {
     new NotebooksListPage(url + path)
   }
 
+  def createFileAtRootDir(googleProject: GoogleProject, clusterName: ClusterName, fileName: String)(implicit token: AuthToken): File = {
```
Comment (Qi77Qi) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/Notebook.scala]: maybe rename this to something like `createFileInJupyter`?
It was weird to me that creating a file at Root dir requires a http call

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookExtensionSpec.scala (hunk 1 above)
Comment (jdcanas) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookExtensionSpec.scala:95]: I don't think we can. I added an additional comment to explain why it's needed. The goal is to check that after 4minutes, the condition holds true. I want to make sure no subsequent periodic ticks mess up the assertions. Eventually exits as soon as it is true. I looked through the ```org.scalatest.concurrent``` documentation and couldn't find any cleaner ways to do this.

**Review by rtitle — APPROVED** [review]
Looks good. Once this is in I can merge it into https://github.com/DataBiosphere/leonardo/pull/1008, which is nearly ready.

**Review by Qi77Qi — APPROVED** [review]

**Pull request merged**